src/azure-cli-core/azure/cli/core/cloud.py

```diff
@@ -192,24 +234,25 @@ def _arm_to_cli_mapper(arm_dict):
-            active_directory_data_lake_resource_id=arm_dict['activeDirectoryDataLake'] if 'activeDirectoryDataLake' in arm_dict else None,  # pylint: disable=line-too-long
-            app_insights_resource_id=arm_dict['appInsightsResourceId'] if 'appInsightsResourceId' in arm_dict else None,
-            log_analytics_resource_id=arm_dict['logAnalyticsResourceId'] if 'logAnalyticsResourceId' in arm_dict else None),  # pylint: disable=line-too-long
+            ossrdbms_resource_id=_get_ossrdbms_resource_id(arm_dict['name']),  # change once ossrdbms_resource_id is available via ARM
+            active_directory_data_lake_resource_id=arm_dict['activeDirectoryDataLake'] if 'activeDirectoryDataLake' in arm_dict else None,
+            app_insights_resource_id=arm_dict['appInsightsResourceId'] if 'appInsightsResourceId' in arm_dict else _get_app_insights_resource_id(arm_dict['name']),
+            log_analytics_resource_id=arm_dict['logAnalyticsResourceId'] if 'logAnalyticsResourceId' in arm_dict else _get_log_analytics_resource_id(arm_dict['name']),
+            app_insights_telemetry_channel_resource_id=arm_dict['appInsightsTelemetryChannelResourceId'] if 'appInsightsTelemetryChannelResourceId' in arm_dict else _get_app_insights_telemetry_channel_resource_id(arm_dict['name'])),
         suffixes=CloudSuffixes(
             storage_endpoint=arm_dict['suffixes']['storage'],
-            storage_sync_endpoint=arm_dict['suffix']['storageSyncEndpointSuffix'] if 'storageSyncEndpointSuffix' in arm_dict['suffixes'] else _get_storage_sync_endpoint(arm_dict['name']),  # pylint: disable=line-too-long
-            keyvault_dns=arm_dict['suffixes']['keyVaultDns'],
-            sql_server_hostname=arm_dict['suffixes']['sqlServerHostname'],
-            mysql_server_endpoint=arm_dict['suffixes']['mysqlServerEndpoint'],
-            postgresql_server_endpoint=arm_dict['suffixes']['postgresqlServerEndpoint'],
-            mariadb_server_endpoint=arm_dict['suffixes']['mariadbServerEndpoint'],
-            azure_datalake_store_file_system_endpoint=arm_dict['suffixes']['azureDataLakeStoreFileSystem'] if 'azureDataLakeStoreFileSystem' in arm_dict['suffixes'] else None,  # pylint: disable=line-too-long
-            azure_datalake_analytics_catalog_and_job_endpoint=arm_dict['suffixes']['azureDataLakeAnalyticsCatalogAndJob'] if 'azureDataLakeAnalyticsCatalogAndJob' in arm_dict['suffixes'] else None,  # pylint: disable=line-too-long
-            acr_login_server_endpoint=arm_dict['suffixes']['acrLoginServer'] if 'acrLoginServer' in arm_dict['suffixes'] else None))  # pylint: disable=line-too-long
+            storage_sync_endpoint=arm_dict['suffix']['storageSyncEndpointSuffix'] if 'storageSyncEndpointSuffix' in arm_dict['suffixes'] else _get_storage_sync_endpoint(arm_dict['name']),
+            keyvault_dns=_add_starting_dot(arm_dict['suffixes']['keyVaultDns']),
```

Comment:
So this changes previous endpoint value(add dot), right ?

Reply:
Yes. Previous endpoint value from ARM metadata is broken.

Actually, from the very beginning, the values may be the same as Azure CLI. Then there's this [PR](https://msazure.visualstudio.com/One/_git/AzureUX-ARM/pullrequest/2296339?_a=files), it removed the starting dots for suffixes in ARM metadata.